src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -216,6 +216,13 @@ export class SelectBox extends vsSelectBox {
 		let selectOptions: SelectOptionItemSQL[] = SelectBox.createOptions(options);
 		this.populateOptionsDictionary(selectOptions);
 		super.setOptions(selectOptions, selected);
+		let selectedOptionIndex = this._optionsDictionary.get(this._selectedOption);
+		if (selectedOptionIndex === selected) { // Not generating an event if the same value is selected.
+			return;
+		}
+		if (this._dialogOptions !== undefined) {
+			this._selectedOption = this._dialogOptions[selected]?.value;
```

Comment:
This seems like a bigger issue to me - why is calling setOptions with the selected value not causing the onSelected event to be fired? I wouldn't expect this change to be necessary if that's behaving like I'd expect it to.

Reply:
So the super.select is the one called and its not setting the this._selectedoption.

Reply:
Doesn't super.select fire the onSelected event though? (which we then have a handler for)

Reply:
sorry super.setOptions(selectOptions, selected); not super.select - no its doesnt fire anything afaik

Reply:
Looked into it and yeah it doesn't. That's not super great since we've modified other things to fire that event when updated programmatically. 

But anyways then this is fine then. We should probably be firing the event in this case too but let's hold off on that until we come across an actual issue with that not being done. @alanrenmsft FYI

Could you update your comment about this though to add a bit more information?

`Calling super.setOptions even with a value selected doesn't fire the onSelected event, so manually update it here to ensure that the value is kept in sync. Currently we're not firing the onDidSelect event in this case to align with the base implementation, but that can change if there's a need for that to happen.`

Reply:
Missed to mention one thing - as the setoption stands currently firing event in it might turn into loop of events (didnt try to change super one but tried with one here).

Reply:
That's what the "don't fire event if the value is the same" should prevent, but yeah stuff like that is why I'm hesitant to change it unless we think we actually need it.